enhancements/microshift/microshift-telemetry.md

```diff
@@ -0,0 +1,284 @@
+  belonging to that specific cluster in the staging environment.
+
+### Sending metrics
+OpenShift is sending metrics through this API every 4.5 minutes. For MicroShift
+deployments this might be a bit excessive due to resource usage, network
+traffic and usage patterns.
+
+Due to the Telemetry API characteristics, MicroShift will send data every hour.
+In terms of resource usage, each metric is about 250B in size and MicroShift
+will send 7 of them. Rounding up because of the multiple time series in the
+metrics measuring pods, resources and usage, each MicroShift telemetry message
+will be about ~2KB.
+
+When MicroShift starts it should send metrics once and then schedule them every
+hour.
+It will also send metrics when issued to stop, as part of the finish process.
+This helps in sending a last report and also hint when the cluster has been
+stopped.
+In order to introduce some entropy to alleviate traffic spikes in case of
+massive deployments (devices in a fleet, for example), the sending interval may
+be added a random offset to throw it off sync. Since the sending interval is
+1h, a 10% offset seems reasonable to spread out sending metrics over 12min.
+This is not configurable.
+
+As described above MicroShift will be using the [direct request](https://github.com/openshift/telemeter/tree/main?tab=readme-ov-file#metricsv1receive-endpoint-receive-metrics-in-prompbwriterequest-format-from-any-client)
+endpoint.
+Each metric must follow [Prometheus WriteRequest](https://github.com/prometheus/prometheus/blob/release-2.38/prompb/remote.proto#L22)
+format.
+
+Some metrics, like those that relate to hardware capabilities (which also
+include the MicroShift version) do not change unless there is a restart. These
+metrics can be sent only when MicroShift starts and/or stops.
+
+### Sensitive data
+There is no user or private data in any of the metrics in MicroShift reports.
+
+### Opting out
+MicroShift may not always have the possibility of sending metrics. There may be
+disconnected clusters, constrained environments where external traffic could be
+audited, customers who simply do not want to share this information, etc. For
+these reasons there must be an opt-out for this functionality, as OpenShift
+already [allows](https://docs.openshift.com/container-platform/4.17/support/remote_health_monitoring/opting-out-of-remote-health-reporting.html).
+
+Taking advantage of the configuration options that the feature requires, an
+enable/disable toggle is provided.
+
+### Workflow Description
+**MicroShift** is the MicroShift main process.
+
+1. MicroShift starts up.
+2. MicroShift reads configuration. If telemetry is not enabled, finish here. If telemetry is enabled proceed to next step.
+3. Collect all metrics and send them. Include dynamic metrics from sampling if available. Retry if failed.
+4. Schedule next send for `telemetry.reportInterval`.
+5. Every `telemetry.sampleInterval` collect all dynamic metrics and store them in memory.
```

Comment:
this seems off now?

Reply:
Fixed it, a leftover.

Reply:
hmm I still fail to find mentions of `sampleInterval` elsewhere. I assume that it's redundant with `reportInterval`.